From c095f1b25ec9bca24cf6a5c0e6699b876b6681aa Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Sat, 14 Feb 2026 02:46:06 +0000
Subject: [PATCH 1/2] cargo: upgrade jsonwebtoken

Move jsonwebtoken to a workspace dependency.
Upgrade to 10 series, to pick up CVE fix in 10.3.0.
Use aws-lc backend, as aws-lc is already an existing transitive
dependency.
---
 Cargo.toml                        | 1 +
 services/azure-storage/Cargo.toml | 2 +-
 services/google/Cargo.toml        | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 8226293..8bdb36d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -52,6 +52,7 @@ hex = "0.4"
 hmac = "0.12"
 http = "1"
 jiff = "0.2"
+jsonwebtoken = { version = "10", features = ["aws_lc_rs"] }
 log = "0.4"
 percent-encoding = "2"
 pretty_assertions = "1.3"
diff --git a/services/azure-storage/Cargo.toml b/services/azure-storage/Cargo.toml
index 4b8ee12..706d27f 100644
--- a/services/azure-storage/Cargo.toml
+++ b/services/azure-storage/Cargo.toml
@@ -41,7 +41,7 @@ serde_json = { workspace = true }
 sha1 = { workspace = true }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-jsonwebtoken = "9.2"
+jsonwebtoken = { workspace = true }
 pem = "3.0"
 rsa = { workspace = true }
 
diff --git a/services/google/Cargo.toml b/services/google/Cargo.toml
index 70e528b..62f6b4d 100644
--- a/services/google/Cargo.toml
+++ b/services/google/Cargo.toml
@@ -30,7 +30,7 @@ rust-version.workspace = true
 async-trait = { workspace = true }
 form_urlencoded = { workspace = true }
 http = { workspace = true }
-jsonwebtoken = "9.2"
+jsonwebtoken = { workspace = true }
 log = { workspace = true }
 percent-encoding = { workspace = true }
 rand = { workspace = true }
-- 
2.51.0

